docs/microphone.rst

```diff
@@ -0,0 +1,83 @@
+Microphone **V2**
+*****************
+
+.. py::module:: microbit
+
+This object lets you access the on-board microphone available on the **V2**
+micro:bit version. It can be used to respond to sound. The microphone input
+is located on the front of the board alongside a microphone activity LED,
+which is lit when the microphone is in use.
+
+.. image:: microphone.png
+    :width: 300px
+    :align: center
+    :height: 240px
+    :alt: micro:bit with microphone LED on
+
+Sound events
+============
+The microphone can respond to a pre-defined set of sound events that are
+based on the amplitude and wavelength of the sound.
+
+These sound events are identified by strings passed as method arguments and
+return values. These strings are also stored as `MicrobitMicrophone` class
+variables as a handy way to identify what events are available.
+
+Classes
+=======
+.. py:class:: MicrobitMicrophone
+
+    .. note::
+        The ``MicrobitMicroPhone`` class documented here can be accessed via 
+        the microbit module as ``microbit.microphone``. It cannot be
+        initialised on it's own.
+
+    Represents the microphone.
+
+.. py:class:: SoundEvent  
+
+    Value to represent loud sound events, like clapping or shouting
+    ``SoundEvent.LOUD`` = ``loud``.
+
+    Value to represent quiet sound events, like speaking or background music
+    ``SoundEvent.QUIET`` = ``quiet``.
+
+.. py:function:: current_event()
```

Comment:
These are methods inside the MicrobitMicrophone class, right now they are rendered as functions.

Looking at the other docs they are not really doing class methods, but list it as functions in `microbit.xxx.foo()`:

![image](https://user-images.githubusercontent.com/29712657/100260852-dc47ba00-2f41-11eb-9430-5dcdd56d13f7.png)

Reply:
I've updated the module link in https://github.com/bbcmicrobit/micropython/pull/692/commits/e1e84874afb910431edbacfae2b3fb27771d3e56 so it is similar to the accelerometer example, hopefully this will render them like the other examples.

Reply:
I think having the MicroBitMicrophone class without any information about it is not really helpful. I think I'd rather either add the functions as methods to the class, or follow the accelerometer example and not include the class and mark all the functions as `microbit.microphone.xxxx`.

Right now and leaning more towards doing it the same way as the accelerometer, thoughts?

Reply:
I agree, for now the accelerometer example is less confusing.

Reply:
It makes sense to do this in this document, since microbit.microphone is the only thing consuming microbit.SoundEvent. 

Let's continue that in a different thread, as the main goal for this comment has been achieved.

Reply:
https://github.com/bbcmicrobit/micropython/pull/692/files#r533629219